src/main/java/frc/robot/subsystems/ClimbSubsystem.java

```diff
@@ -15,35 +16,43 @@
 public class ClimbSubsystem extends SubsystemBase {
   // Put methods for controlling this subsystem
   // here. Call these from Commands.
-
+  private Joystick m_climbJoystick;
   private MotorController m_climbMotorControllerOne;
   private MotorController m_climbMotorControllerTwo;
   private DigitalInput m_limitSwitch;
+  private double climbHight;
+  private boolean toggleClimb;
 
   // TODO: maybe add servos
 
-  public ClimbSubsystem() {
+  public ClimbSubsystem(Joystick joystick) {
+    m_climbJoystick = joystick;
+    toggleClimb = false;
+    climbHight = 0;
+
     // One is left, two is right
     m_climbMotorControllerOne =
         new MotorController("Climb Motor One", Constants.ClimbMotorOne, 40, true);
     m_climbMotorControllerTwo =
         new MotorController("Climb Motor Two", Constants.ClimbMotorTwo, 40, true);
     m_climbMotorControllerTwo.setInverted(true);
     m_climbMotorControllerTwo.setFollow(m_climbMotorControllerOne);
+    m_climbMotorControllerOne.getEncoder().setPosition(0);
 
     m_limitSwitch = new DigitalInput(Constants.LimitSwitchChannel);
   }
 
+  public void toggleClimbEnable() {
+    toggleClimb = !toggleClimb;
+  }
+
   public void enableClimb(boolean on, boolean up) {
-    if (on) {
-      if (up) {
-        m_climbMotorControllerOne.getEncoder().setPosition(0);
-        m_climbMotorControllerOne.getPID().setReference(25, CANSparkMax.ControlType.kPosition);
-
-      } else {
-        m_climbMotorControllerOne.getEncoder().setPosition(0);
-        m_climbMotorControllerOne.getPID().setReference(-25, CANSparkMax.ControlType.kPosition);
-      }
+    if (on && !m_limitSwitch.get()) {
+      m_climbMotorControllerOne
+          .getPID()
+          .setReference(
+              climbHight + m_climbJoystick.getRawAxis(Constants.leftJoystickY),
```

Comment:
climbHeight is never set to anything besides initialization to zero. Is it means to be keeping track of where we are in the climb?